### DISCUSSION TITLE
How to build ORT offline

### DISCUSSION THREAD

**Discussion by mudream4869** [microsoft/onnxruntime#16071] [Other Q&A] [answered]
Before version 1.14, I could simply clone the project recursively and create the tarball. Then, I could cmake it without requiring network access.

However, starting from version 1.14, it appears that even if I clone the entire project recursively, it still requires downloading something during the cmake process.

I'm not certain if there's any flag or method available to avoid this.

Appreciate any insights~ 😄

**Comment by snnn** [comment, marked as answer]
Download all the things in cmake/deps.txt, then run 
https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/replace_urls_in_deps.py

cmake/deps.txt is a CSV-like file. The second column is the URLs.

**Reply by mudream4869** [reply]
It appears that Eigen is not included in the `deps.txt` file, and the `eigen.cmake` file is utilizing a raw URL instead of a variable loaded from `deps.txt`. Will Eigen be included in the `deps.txt` file?

**Reply by snnn** [reply]
Sure. Would you like to help update https://github.com/microsoft/onnxruntime/blob/main/cmake/external/eigen.cmake? 

It should be in way like:
```
FetchContent_Declare(
    re2
    URL ${DEP_URL_re2}
    URL_HASH SHA1=${DEP_SHA1_re2}
    FIND_PACKAGE_ARGS NAMES re2
)
```
Just replace `re2` to `eigen`.
deps.txt already has eigen.

**Reply by mudream4869** [reply]
👍 Sure, I'm happy to help!
